src/typegenTypeHelpers.ts

```diff
@@ -157,33 +157,20 @@ export type GetGen<K extends GenTypesShapeKeys, Fallback = any> = NexusGen exten
 
 export type GetGen2<
   K extends GenTypesShapeKeys,
-  K2 extends keyof GenTypesShape[K]
-> = NexusGen extends infer GenTypes
-  ? GenTypes extends GenTypesShape
-    ? K extends keyof GenTypes
-      ? K2 extends keyof GenTypes[K]
-        ? GenTypes[K][K2]
-        : any
-      : any
-    : any
-  : any
+  K2 extends Extract<keyof GenTypesShape[K], string>,
+  Fallback = any
+> = K2 extends keyof GetGen<K, never> ? GetGen<K>[K2] : Fallback
```

Comment:
Don't ask me the _exact_ reason why this is fixing anything. I assume this is due to some TS internals (which may backfire to us later...) and the way they compute union distribution. Perhaps in this case the computation is deferred, perhaps this has no relation at all and I'm saying bs.

A bit of explanation though: We intentionally use `GetGen<K, never>` so that if `GetGen<K>` fails, we fallback to `never` which ensures that `K2 extends of never` will never be true, thus falling in the false case.